packages/react-api/src/Api.tsx

```diff
@@ -81,24 +80,64 @@ function getDevTypes (): Record<string, Record<string, string>> {
   return types;
 }
 
-async function getInjectedAccounts (injectedPromise: Promise<InjectedExtension[]>): Promise<InjectedAccountExt[]> {
+async function getInjectedAccounts (extensions: InjectedExtension[]): Promise<InjectedAccountExt[]> {
+  const promisedAccounts = extensions.map((extension) => extension.accounts.get());
+
   try {
-    await injectedPromise;
+    const fetchedAccounts = await Promise.all(promisedAccounts);
+
+    let whenCreated = 0;
+
+    return extensions.flatMap((extension, index) => (
+      fetchedAccounts[index].map((fetchedAccount) => (
+        {
+          address: fetchedAccount.address,
+          meta: {
+            name: `${fetchedAccount.name || 'unknown'} (${extension.name})`,
+            source: extension.name,
+            whenCreated: whenCreated++
+          }
+        }
+      ))
+    ));
+  } catch (error) {
+    console.error('web3Accounts', error);
 
-    const accounts = await web3Accounts();
+    return [];
+  }
+}
 
-    return accounts.map(({ address, meta }, whenCreated): InjectedAccountExt => ({
-      address,
-      meta: objectSpread({}, meta, {
-        name: `${meta.name || 'unknown'} (${meta.source === 'polkadot-js' ? 'extension' : meta.source})`,
-        whenCreated
-      })
-    }));
+async function getExtensions (): Promise<InjectedExtension[]> {
+  const injectedWindow = window as Window & InjectedWindow;
+  const promisedExtensions: Promise<Injected>[] = [];
+  let enabledExtensions: Injected[] = [];
+  const extensionsInfo: InjectedExtensionInfo[] = [];
+  const extensionsWithInfo: InjectedExtension[] = [];
+
+  const supportedExtensions = supportedExtensionsNames().filter((supportedExtension) => !DISALLOW_EXTENSIONS.includes(supportedExtension));
+  const storageData = localStorage.getItem('enabledExtensions');
+  const storedEnabledExtensions: string[] | null = storageData !== null ? JSON.parse(storageData) as string[] : null;
+
+  Object.keys(injectedWindow.injectedWeb3).forEach((extensionName) => {
```

Comment:
This is really problematic - we are no only allowing 2 extensions to operate, since -

1. the one extension is ours
2. the second extension we seem to be playing favorites to

This means that anything else would be skipped. I don't want to go down the route of playing favorites or playing traffic-warned to which should be allowed. I'm 100% on-board that when teams have added their extension here, it has special treatment -

- allows links to install,
- it shows up properly configured,
- etc.

However if a team has not done that, I really don't want to disallow operation outright.

Reply:
A solution that can be both fair and UX friendly would be to have the other injected extentions that are not added with full info under Others section. The full list of injected extensions are available in windows.injectedWeb3, so for regsitered extensions we can show the full info, for unrgistered extensions you can still get the name and version and show them in the dropdown list as an "Others" section. We still should have the DISALLOW_EXTENSIONS IMO to be able to filter the sketchy/scammy extensions.